datastore-design/go-grpc-demo/storagebackend/timescaledb/findtimeseries.go

```diff
@@ -0,0 +1,143 @@
+package timescaledb
+
+import (
+	"datastore/datastore"
+	"fmt"
+	"strings"
+
+	postgis "github.com/cridenour/go-postgis"
+	_ "github.com/lib/pq"
+)
+
+// matchAnyValCond creates the sub-condition of a WHERE clause that checks if string column 'name'
+// matches any of a set of values. Placeholder values are appended to phVals.
+// Returns sub-condition (with placeholders) to be appended to a WHERE clause.
+func matchAnyValCond(phVals *[]interface{}, name string, values []string) string {
+
+	if len(values) == 0 {
+		return ""
+	}
+
+	inArgs := []string{}
+	phIndex := len(*phVals) + 1 // ensure placeholders start at $1, not $0
+	for _, value := range values {
+		inArgs = append(inArgs, fmt.Sprintf("$%d", phIndex))
+		*phVals = append(*phVals, value)
+		phIndex++
+	}
+
+	return fmt.Sprintf(" AND %s IN (%s)", name, strings.Join(inArgs, ","))
+}
+
+func equal(p1, p2 *datastore.Point) bool {
+	return (p1.Lat == p2.Lat) && (p1.Lon == p2.Lon)
+}
+
+// insidePolygonCond creates a sub-condition of a WHERE clause that checks if geo point column
+// 'name' is contained in a geo polygon.
+// Returns (the sub-condition to be appended to a WHERE clause, nil) upon success, otherwise
+// (..., error).
+func insidePolygonCond(name string, polygon0 *datastore.Polygon) (string, error) {
+
+	if polygon0 == nil {
+		// absent polygon => disable filtering (note: "" is equivalent to " AND TRUE")
+		return "", nil
+	}
+
+	points := polygon0.Points
+
+	// validate polygon and close it if necessary
+	if len(points) < 3 {
+		return "", fmt.Errorf("polygon must contain at least 3 points; found %d", len(points))
+	}
+	if equal(points[0], points[len(points) - 1]) { // endpoints equal -> polygon already closed
+		if len(points) == 3 {
+			return "", fmt.Errorf(
+				"polygon must contain at least 4 points when endpoints are equal; found %d",
+				len(points))
+		}
+	} else { // endpoints different, so close polygon
+		points = append(points, points[0])
+	}
```

Comment:
This works of course, and I don't know how much we want to discuss the code. But... I think the code is a bit cleaner if you first do the append if if it not closed, and then check for length. Or did you choose this approach so you can have separate error message for closed and non-closed polygons?

Reply:
Good point. I agree that code simplicity has priority over error message precision in this case. Update coming up.